Adds Syndicate organizations

datlo opened the pull request:
<!-- By ticking or leaving ticked the option "Allow edits and access to secrets by maintainers" you give permission for repository maintainers to push changes to your branch without explicitly asking. -->

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
<!-- Include a small to medium description of what your PR changes. -->
<!-- Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->
<!-- If your PR fixes an issue, add "Fixes #1234" somewhere in the PR description. This will automatically close the bug upon PR submission. -->

This PR adds Syndicate lore organizations. Each traitor player gets assigned an org that influences objectives and steal targets according to the organization's theme. Some organizations are laser focused on their goals like the Animal Rights Consortium who will always go after animal abusers, while others may pick up odd jobs unrelated to their targets of choice occasionally. This prevents antags from being *too* predictable.

Hijack chance is not affected by this PR, with hijackers getting their own organization.

Significant balance change from this is that 10 percent of syndicate agents will target other syndicate agents, knowing that they are syndicate. Hunter targets will receive a notification that they are being hunted. Other syndicate agents have a 5 percent chance of getting that notification anyway to keep them on their toes.


<details>
<summary>
<i>Organization list</i>
</summary>
<p>

- Donk Co : Default traitor, random objectives.
- Hawkmoon Acquisitions : Theft only
- Animal Rights Consortium : Teach a lesson to animal abusers (Onstation Cargo/Service/Genetics/Virology)
- Waffle Company : Assassination and variants only
- Cybersun Incorporated : Mostly target command/security
- Interdyne Pharmaceuticals : Mostly target medical
- Silicon Engine Liberation Front : Mostly target science
- Electra Dynamics : Mostly target engineering
- Spider Clan Initiate : Target one syndicate agent, one non-mindshielded
- FAID : Target one syndicate agent, steal station blueprints/docs
- Gorlex Marauders : Hijack only

</p>
</details>

## Why It's Good For The Game
<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
This PR makes lore have an impact on the traitor's motive/backstory, giving players a direction on where to take their roleplay for their antag shenanigans. This adds flavor to traitor gameplay without significantly impacting game balance.

In the future, I would like to modify organization distribution to be based off the amount of players and security players in the round. Less sec = Less violent antagonists. As this would be a large project, I'm reserving it for an eventual future PR. This starts framework for that, though.


## Images of changes
<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif or mp4 of your feature if you want. -->
![image](https://github.com/user-attachments/assets/eff97169-9319-4bad-aaa4-bb6c2007176a)
Syndicate agents targeted by syndicate hunters warning on objective reveal :
![image](https://github.com/user-attachments/assets/48d51f5d-4042-4d18-8a7d-58df00d5a054)

## Testing
<!-- How did you test the PR, if at all? -->
Tested all orgs and methods of traitor spawning.

## Todo

- [x] Theft targets
- [x] Faction specific objectives
- [x] Lore writing
- [x] Testing

## Changelog
:cl: Pi and the Lore Team
add: Added syndicate organizations for traitors.
add: Traitor objectives are now influenced by the organization you're affiliated with.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

Coolrune206 commented:
I don't favour selecting organisations to get a higher chance of specific objectives and discounts because then people can just select Gorlex for instance and always have very loud objectives with discounts well suited for them.

datlo commented:
> I don't favour selecting organisations to get a higher chance of specific objectives and discounts because then people can just select Gorlex for instance and always have very loud objectives with discounts well suited for them.

I understand the concern, and this will be considered when I code the picking process, but essentially picking Gorlex marauders as your favorite will not make it more likely that you get Gorlex over any other choice, it only makes it so that if the algo needs to decide between Gorlex and another org, Gorlex will be the first choice. As gorlex is a high chaos org, it'll only be seen when security is high in numbers compared to antags. The favorite faction will matter a lot more for less chaotic orgs which have a lot of competition between them for the slot.

DGamerL commented on code/game/gamemodes/objective.dm:
```diff
@@ -117,9 +124,21 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
+			continue
 		possible_targets += possible_target
 
+	if(!possible_targets.len) //If we can't find anyone, try with less restrictions
```
```suggestion
	if(!length(possible_targets)) //If we can't find anyone, try with less restrictions
```

DGamerL commented on code/game/gamemodes/objective.dm:
```diff
@@ -117,9 +124,21 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
+			continue
+		if(syndicate_target && possible_target.special_role != SPECIAL_ROLE_TRAITOR)
+			continue
+		if(target_jobs.len && !(possible_target.assigned_role in target_jobs))
```
```suggestion
		if(length(target_jobs) && !(possible_target.assigned_role in target_jobs))
```

DGamerL commented on code/game/gamemodes/objective.dm:
```diff
@@ -177,7 +196,6 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 		return FALSE
 	return TRUE
-
 /datum/objective/assassinateonce
```
```suggestion

/datum/objective/assassinateonce
```
Can probably keep this newline otherwise it's too close together

DGamerL commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -135,19 +140,23 @@
-		else
-			add_antag_objective(/datum/objective/assassinate)
+	//If our org has an objectives list, give one to us if we pass a roll on the org's focus
+	if(organisation && organisation.objectives.len && prob(organisation.focus))
```
```suggestion
	if(organisation && length(organisation.objectives) && prob(organisation.focus))
```

lewcc commented on code/modules/antagonists/antag_org/antag_org_syndicate.dm:
```diff
@@ -0,0 +1,124 @@
+	intro_desc = "(TODO - LORE) YOU IS VOXXY LOOTING CLUB! SKREEE GO GO GO WHAT IS YOU WAITINGS FOR?!"
+	focus = 100
+	objectives = list(/datum/objective/steal)
+	difficulty = "Easy"
```
You should probably create defines for these difficulty levels

lewcc commented on code/modules/antagonists/antag_org/antag_org_datum.dm:
```diff
@@ -0,0 +1,34 @@
+	/// Used for prob() for objectives. Higher focus means the org is less likely to diverge from their favorites.
+	var/focus = 100
+	/// If set, the antag's first objective will be forced to this.
+	var/forced_objective
```
```suggestion
	var/datum/objective/forced_objective
```

lewcc commented on code/modules/antagonists/antag_org/antag_org_syndicate.dm:
```diff
@@ -0,0 +1,124 @@
+	name = "Steal Item (ANARCHPRIM)"
+	steal_list = list(/datum/theft_objective/supermatter_sliver, /datum/theft_objective/nukedisc, /datum/theft_objective/plutonium_core)
+
+/datum/antag_org/syndicate/gorlex
```
If you add focuses, you should *probably* exclude this one

DGamerL replied:
This one is going to be only assigned to people that roll hijack IIRC

datlo replied:
Correct

DGamerL commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -106,21 +110,22 @@
-	// Will give normal steal/kill/etc. type objectives.
-	for(var/i in 1 to GLOB.configuration.gamemode.traitor_objectives_amount)
-		forge_single_human_objective()
+	var/i = 0
```
Could you rename this variable?

DGamerL commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -106,21 +110,22 @@
 
 	// Give them an escape objective if they don't have one already.
-	if(!(locate(/datum/objective/escape) in owner.get_all_objectives()) && (!can_succeed_if_dead || prob(80)))
+	if(!(locate(/datum/objective/escape) in owner.get_all_objectives()))
```
Undocumented change.

datlo replied:
This is leftover code from when we had the martyr objective as an option instead of escape, there's no change incurred from removing it.

Komrad822 commented:
So an idea, what if you could... disable certain factions. But the caviate is you cant disable any if you are Donk co and you can only disable harder factions, not easier ones. So If you enable Gorlex you cant disable Interdyne or Waffle Co. but if you choose Waffle co, you can disable normal difficulty factions and up. 

This is so you can either never get super hard objectives or you keep getting nad objective one after the other and you just want a break without turning off traitor.

S34NW commented:
No movement in a month, please update us on your plan for this or work on it further in the next week or we will have to close it for inactivity.

Pedritoc25 commented:
Would be great see this in game

datlo commented:
I've been unfortunately busy lately, I would still like to do this but I have to put it on hold for now.

If anyone would like to pick the project back up, feel free, ask if you got any questions.

If not, I will eventually come back to this once I can.

datlo commented:
This is now ready for review, although I still need lore written to finish this proper. Code is done unless I find/am told of any issues.

Sirryan2002 commented:
Can this PR be converted to a WIP state until you're finished with everything in it? Including lore snippets

datlo commented:
Finally ready for review

Pedritoc25 commented on code/modules/antagonists/antag_org/antag_org_syndicate.dm:
```diff
@@ -81,15 +92,9 @@
-/datum/antag_org/syndicate/waffle
-	name = "Waffle Company"
-	intro_desc = "(TODO - LORE) You are a Waffle Company operative. Get in, fuck shit up, get out with a fancy new shuttle. You know the drill."
+	intro_desc = "(TODO - LORE) You are an operative of the infamous Gorlex Marauders, a brutal and merciless gang of pirates and cutthroats. \
```
Hi, you left a "(TODO - LORE)" in the final Gorlex Traitor Type

datlo replied:
This is fixed, nto sure why it wasnt there.

Sirryan2002 commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -82,6 +82,16 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
+	var/chaos = pick(prob(ORG_PROB_HUNTER);ORG_CHAOS_HUNTER, prob(ORG_PROB_MILD);ORG_CHAOS_MILD, prob(ORG_PROB_AVERAGE);ORG_CHAOS_AVERAGE, prob(ORG_PROB_HIJACK);ORG_CHAOS_HIJACK)
+	var/org_list = shuffle(subtypesof(/datum/antag_org/syndicate))
+	for(var/org_type in org_list)
+		var/datum/antag_org/org = new org_type(src)
+		if(org.chaos_level == chaos)
+			organisation = org
+			break
+		qdel(org)
```
can't you check the chaos level of the org using `initial()` without have to new() that type?

datlo replied:
I'm not sure how I'd go about doing that, do you have an example on the codebase?

Sirryan2002 replied:
```dm
	for(var/path in subtypesof(/datum/keybinding))
		var/datum/keybinding/D = path
		if(initial(D.name))
			GLOB.keybindings += new path()
```

datlo replied:
Thanks! Done and tested, works

Sirryan2002 commented on code/game/gamemodes/objective.dm:
```diff
@@ -527,6 +545,7 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	var/datum/theft_objective/steal_target
 	martyr_compatible = FALSE
 	var/theft_area
+	var/list/steal_list = list()
```
please document this variable :)

Contrabang commented on code/game/gamemodes/objective.dm:
```diff
@@ -32,7 +32,14 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
-
+	/// List of jobs that the objective will target if possible, any crew if not.
+	var/list/target_jobs = list()
+	/// If set to TRUE, this objective will target mindshielded crew if possible, any crew if not.
+	var/mindshielded_target = FALSE
+	/// If set to TRUE, this objective will target non-mindshielded crew if possible, any crew if not.
+	var/not_mindshielded_target = FALSE
+	/// If set to TRUE, this objective will target a syndicate agent if possible, any crew if not.
+	var/syndicate_target = FALSE
```
This should use flags instead, and a single variable

Contrabang commented on code/game/gamemodes/objectives_subtypes.dm:
```diff
@@ -0,0 +1,58 @@
+	name = "Assassinate engineering"
+	target_jobs = list("Chief Engineer", "Station Engineer", "Atmospherics Technician")
+
+/datum/objective/assassinateonce/animal_abuser
+	name = "Assassinate animal abuser once"
+	target_jobs = list("Head of Personnel", "Quartermaster", "Cargo Technician", "Bartender", "Chef", "Botanist", "Geneticist", "Virologist")
+
+/datum/objective/assassinateonce/animal_abuser/update_explanation_text()
+	..()
+	if(target?.current)
+		explanation_text = "Teach [target.current.real_name], the animal abuser and alleged [target.assigned_role], a lesson they will not forget. The target only needs to die once for success."
```
Personal opinion but I really don't like this or think we should have this as an objective.

datlo replied:
Is the issue a gameplay one, or the subject matter?

Contrabang replied:
IMO subject matter. We've removed similar abductor objectives iirc

lewcc replied:
Yeah, I think this could probably be reworded if nothing else; calling people's characters animal abusers is a little icky.

datlo replied:
Removed 'abuser' vocabulary from the code, description and the objective explanation. Now looks like a regular teach a lesson objective, still targeting the same jobs.

lewcc replied:
It makes sense that it's written from the perspective of a PETA like org, but yeah like I wouldn't want my own characters characterized like that, nor have someone take the objective a bit too seriously and start screaming about that ic.

Contrabang commented on code/game/gamemodes/objectives_subtypes.dm:
```diff
@@ -0,0 +1,58 @@
+/datum/objective/debrain/command
+	name = "Debrain command"
+	target_jobs = list("Captain", "Head of Personnel", "Head of Security", "Research Director", "Chief Engineer", "Chief Medical Officer", "Quartermaster")
```
This would be much more sane if this used a departmental strings or departmental flags, and then used a global proc to get job list from global lists like `command_head_positions`

Contrabang commented on code/modules/antagonists/_common/antag_datum.dm:
```diff
@@ -41,6 +41,8 @@ GLOBAL_LIST_EMPTY(antagonists)
 	/// The url page name for this antagonist, appended to the end of the wiki url in the form of: [GLOB.configuration.url.wiki_url]/index.php/[wiki_page_name]
 	var/wiki_page_name
+	/// The organisation, if any, this antag is associated with
+	var/datum/antag_org/organisation
```
I don't think this should be on all antags, just traitors. Just so things are properly encapsulated. If someone wants to add this to clings/vampires, they can move it down themselves

datlo replied:
Moving this down to the traitor antag would require refactoring the on_gain proc so the introduction message is shown properly to an unnecessary extent.

hal9000PR replied:
I think this could be made use of by more than just traitors.
We could add vampire cabals to the lore, cling dynasties, wizard schools, etc.

Burzah replied:
Off topic here, but since this is on the same line. Variables declared should use American English as the spelling, in accordance with `CONTRIBUTING.md`

datlo replied:
All instances have been rewritten to 'Organization' per the guidelines, thank you for pointing this out!

Contrabang commented on code/modules/antagonists/antag_org/antag_org_syndicate.dm:
```diff
@@ -0,0 +1,100 @@
+	focus = 70
+	objectives = list(/datum/objective/assassinate/medical, /datum/objective/assassinateonce/medical, /datum/objective/steal/interdyne)
+
+/datum/objective/steal/interdyne
+	name = "Steal Item (Interdyne)"
+	steal_list = list(/datum/theft_objective/hypospray, /datum/theft_objective/defib, /datum/theft_objective/krav)
+
```
Put these objectives in another file, not mixed in with the antag organization things

Contrabang commented on code/modules/antagonists/antag_org/antag_org_syndicate.dm:
```diff
@@ -0,0 +1,100 @@
+	name = "Steal Item (Cybersun)"
+	steal_list = list(/datum/theft_objective/antique_laser_gun, /datum/theft_objective/nukedisc, /datum/theft_objective/hoslaser)
+
+/datum/antag_org/syndicate/interdyne //Mostly target Medical
```
All your comments should have spaces after the //
```suggestion
/datum/antag_org/syndicate/interdyne // Mostly target Medical
```

Contrabang commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -82,6 +82,14 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
 	return ..()
 
+/datum/antagonist/traitor/select_organisation()
+	var/chaos = pick(prob(ORG_PROB_HUNTER);ORG_CHAOS_HUNTER, prob(ORG_PROB_MILD);ORG_CHAOS_MILD, prob(ORG_PROB_AVERAGE);ORG_CHAOS_AVERAGE, prob(ORG_PROB_HIJACK);ORG_CHAOS_HIJACK)
```
We are trying to move away from this `pick(2;"thing")` dynamic. Consider using something like `pickweight` instead.

Contrabang commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -82,6 +82,14 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
+		var/datum/antag_org/org = org_type
+		if(initial(org.chaos_level) == chaos)
+			organisation = new org_type(src)
+			break
```
```suggestion
			return
```

Contrabang commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -106,23 +114,28 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
+		return //Hijackers only get hijack.
+
+	// Will give objectives from our org or random objectives.
+	while(iteration < GLOB.configuration.gamemode.traitor_objectives_amount)
+		forge_single_human_objective()
+		iteration++
```
Try something like this instead. We prefer for loops over while loops
```suggestion
	for(var/i in iteration to GLOB.configuration.gamemode.traitor_objectives_amount)
		forge_single_human_objective()
```

Contrabang commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -138,25 +151,31 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
+	//If our org has an objectives list, give one to us if we pass a roll on the org's focus
+	if(organisation && length(organisation.objectives) && prob(organisation.focus))
+		add_antag_objective(pick(organisation.objectives))
 	else
```
use an early return instead of an else here

datlo commented:
Thanks for the review, will adress sometime this week

Sirryan2002 commented:
@datlo have you gotten a chance to look at some of this stuff?

datlo commented:
Life has been busy but this will get looked at and adressed

datlo commented:
Ready for review

hal9000PR commented on code/modules/antagonists/antag_org/antag_org_datum.dm:
```diff
@@ -0,0 +1,16 @@
+ /**
+  * The lore organisation antagonists are attached to. Influences objectives and steal targets.
+  */
+/datum/antag_org
+	/// Organisation's name
+	var/name = "Generic Bad Guys"
+	/// Description given to the antagonist on spawn, below 'You are a Traitor!' or similar
+	var/intro_desc = "You are not meant to see this. Please tell admins/coders that the antag_org wasn't set properly."
```
might be good to have both the name and the intro_desc be a debug comment?

hal9000PR commented on code/modules/antagonists/antag_org/antag_org_datum.dm:
```diff
@@ -0,0 +1,16 @@
+	var/intro_desc = "You are not meant to see this. Please tell admins/coders that the antag_org wasn't set properly."
+	/// Used for prob() for objectives. Higher focus means the org is less likely to diverge from their favorites.
+	var/focus = 100
+	/// If set, the antag's first objective will be forced to this.
+	var/datum/objective/forced_objective
```
might this be worth being a list? incase we want to force multiple objectives.

DGamerL replied:
Making this a list wouldn't be the worst thing, I agree

hal9000PR commented on code/game/gamemodes/objectives_subtypes.dm:
```diff
@@ -0,0 +1,73 @@
+/datum/objective/debrain/science
+	name = "Debrain science"
+	target_department = DEPARTMENT_SCIENCE
```
instead of having a million overrides for murder/engineer, murder/science, murder/sec, etc. wouldnt it be better to have a proc on datum/antag_org that creates these objectives and overrides these variables?

datlo replied:
Yeah I'll rewrite how this works to shorten lines before merge

DGamerL commented on code/game/gamemodes/objectives_subtypes.dm:
```diff
@@ -0,0 +1,68 @@
+	target_department = DEPARTMENT_ENGINEERING
+
+/datum/objective/assassinateonce/animal_user
+	name = "Assassinate animal user"
```
... assasinate animal user? Do they do lines of cokepybaras?

Scribble-Sheep replied:
Maybe animal product user?

DGamerL replied:
(I think they meant *ab*user)

datlo replied:
This is admin tools/code viewable only but sure ill change it to that

datlo commented:
Rewritten to the US spelling 'Organization' per the CONTRIBUTING.md guidelines

datlo commented:
Refactored the objective selection process to remove the dozens of subtypes. 

Reworked it while I was at it to adress concerns I got that assassinations were too high on the test merge : 
Syndicate objectives code now use the same objective distribution as previously if your organisation does not have specific objectives such as steal, hijack, etc.

This should result in a smaller increase in round removal objectives.

Burzah commented on code/modules/antagonists/traitor/datum_traitor.dm:
```diff
@@ -104,23 +112,31 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
+	if(organization && length(organization.forced_objectives))
+		for(var/forced_objectives in organization.forced_objectives)
+			var/datum/objective/forced_obj = forced_objectives
+			if(!ispath(forced_obj, /datum/objective/hijack) && delayed_objectives) //Hijackers know their objective immediately
```
```suggestion
			if(!ispath(forced_obj, /datum/objective/hijack) && delayed_objectives) // Hijackers know their objective immediately
```

Burzah commented on code/game/gamemodes/objective.dm:
```diff
@@ -122,9 +131,22 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
+			continue
 		possible_targets += possible_target
 
+	if(!length(possible_targets)) //If we can't find anyone, try with less restrictions
```
```suggestion
	if(!length(possible_targets)) // If we can't find anyone, try with less restrictions
```

Burzah reviewed (approved):
Looks good to me. No major issues have been reported during the test merge. Recent adjustment to the percentage of assassination objectives seems yo have brought in in-line to what it already was.

Sirryan2002 commented on code/game/gamemodes/objective.dm:
```diff
@@ -122,9 +131,22 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 		if(is_invalid_target(possible_target) || (possible_target in target_blacklist))
 			continue
-
+		if(flags_target & MINDSHIELDED_TARGET && !ismindshielded(possible_target.current))
+			continue
+		if(flags_target & UNMINDSHIELDED_TARGET && ismindshielded(possible_target.current))
+			continue
+		if(flags_target & SYNDICATE_TARGET && possible_target.special_role != SPECIAL_ROLE_TRAITOR)
+			continue
```
```suggestion
		if((flags_target & MINDSHIELDED_TARGET) && !ismindshielded(possible_target.current))
			continue
		if((flags_target & UNMINDSHIELDED_TARGET) && ismindshielded(possible_target.current))
			continue
		if((flags_target & SYNDICATE_TARGET) && possible_target.special_role != SPECIAL_ROLE_TRAITOR)
			continue
```
While this works without parentheses, wrapping bit operations helps with readability

Sirryan2002 commented on code/modules/antagonists/antag_org/antag_org_syndicate.dm:
```diff
@@ -0,0 +1,84 @@
+		Clean kills, clean thefts, clean getaway. Get it done, operative."
+	focus = 50 // Don't bully sec too hard
+	targeted_departments = list(DEPARTMENT_COMMAND, DEPARTMENT_SECURITY)
+	theft_targets = list(/datum/theft_objective/antique_laser_gun, /datum/theft_objective/nukedisc, /datum/theft_objective/hoslaser, \
```
maybe consider converting this in a multiline list definition?

datlo replied:
Is there any specific pro to doing that? Or just a formatting thing

DGamerL replied:
Formatting thing

Burzah replied:
I made a suggestion, and Steel pushed it in. We are good.

Sirryan2002 reviewed (approved):
only a few nitpicks, otherwise code is good

Adrer commented on code/game/gamemodes/objectives_subtypes.dm:
```diff
@@ -0,0 +1,20 @@
+/datum/objective/assassinate/syndicate/update_explanation_text()
+	..()
+	if(target?.current)
+		explanation_text = "Assassinate [target.current.real_name], the Syndicate agent undercover as the [target.assigned_role]."
```
I am personally not a fan of this (Especially since it's not documented in the PR description, please do add it.) I don't think traitors should automatically be aware if their target is a traitor; Currently, as an agent, you can choose not to use your codewords, and stay hidden, meaning that you are treated differently as a target than when you are a known traitor.

datlo replied:
This only applies to hunter syndicates, who are 10 percent of syndicate agents, not all syndies. PR description has been updated to be more explicit on this, this was documented but unclear.

Scribble-Sheep commented:
I'm not sure why the undocumented change was made to tell you when your target's a syndie but I don't like it.
Feels like all it does is just up your chance of getting cheese tactics like sleepy pen used on you more than anything, since they know you'll have tools to fight back.

datlo commented:
> I'm not sure why the undocumented change was made to tell you when your target's a syndie, but I don't like it. Feels like all it does is just up your chance of getting cheese tactics like sleepy pen used on you more than anything, since they know you'll have tools to fight back.

This only applies to hunter syndicates, who are 10 percent of syndicate agents, not all syndies. PR description has been updated to be more explicit on this, this was documented but unclear.

datlo commented:
Syndicate agents targeted by hunters now have a warning on spawn. There's also a low 5 percent chance that you'll get the message even if you aren't targeted.

![image](https://github.com/user-attachments/assets/9fd750f0-e2e8-48f1-ad14-5c4d7a828e55)

1080pCat commented:
A big issue i have with this pr is the fact you get the the same type of objective most of the time. The variation of what types of objectives you needed todo brought unique builds to the table, rather than just "okay go teach x a lesson, then also z one too. 

And telling agents that they're a target is just.. bad. It'll lead to them doing their objectives, then vanishing into deepspace for the rest of the round as to not die.

Scribble-Sheep commented:
> A big issue i have with this pr is the fact you get the the same type of objective most of the time. The variation of what types of objectives you needed todo brought unique builds to the table, rather than just "okay go teach x a lesson, then also z one too.
> 
> And telling agents that they're a target is just.. bad. It'll lead to them doing their objectives, then vanishing into deepspace for the rest of the round as to not die.

The alternative of tots being sleepy penned without a chance to fight specifically back bc they know you have the ability to, isn't much better either, tbh.

1080pCat commented:
> The alternative of tots being sleepy penned without a chance to fight specifically back bc they know you have the ability to, isn't much better either, tbh.

by that logic crew should also be let known that they're a target.

Scribble-Sheep commented:
> > The alternative of tots being sleepy penned without a chance to fight specifically back bc they know you have the ability to, isn't much better either, tbh.
> 
> by that logic crew should also be let known that they're a target.

The logic is that people who know their target is a tot tend to go more for cheese tactics than for civilian targets, and unlike before, there's no way to avoid that by not revealing codewords.

datlo commented:
Any feedback from balance on this? Has been TM'd a while. If the hunter/hunted balance is an issue, Id be in favor of scrapping the feature entirely to stay closer to original traitor objectives.

I've also read feedback that people dont like how similar the objectives feel now, this can be tweaked using prob numbers. Let me know.

lewcc reviewed (approved):
Code looks good to me, beyond outstanding comments.

Burzah commented on code/modules/antagonists/antag_org/antag_org_syndicate.dm:
```diff
@@ -0,0 +1,84 @@
+		Clean kills, clean thefts, clean getaway. Get it done, operative."
+	focus = 50 // Don't bully sec too hard
+	targeted_departments = list(DEPARTMENT_COMMAND, DEPARTMENT_SECURITY)
+	theft_targets = list(/datum/theft_objective/antique_laser_gun, /datum/theft_objective/nukedisc, /datum/theft_objective/hoslaser, \
+		/datum/theft_objective/captains_rapier, /datum/theft_objective/capmedal)
```
```suggestion
	theft_targets = list(
        /datum/theft_objective/antique_laser_gun, 
        /datum/theft_objective/nukedisc, 
        /datum/theft_objective/hoslaser,
		/datum/theft_objective/captains_sabre, 
        /datum/theft_objective/capmedal
    )
```
list format, and typepath update for the change to captain's sword

BAGELMENSK commented:
I like knowing my target is a syndie, I've used that as leverage to RP with crew multiple times to get them to help me, and later to set up dramatic tension for when I take out my fellow baddie. Very fun
